prow/cmd/hook/main.go

```diff
@@ -111,6 +124,12 @@ func main() {
 		if err != nil {
 			logrus.WithError(err).Fatal("Error getting kube client.")
 		}
+
+		if *dry || teamToken == "" {
+			slackClient = nil
```

Comment:
The zero value for a pointer is already `nil`, so you can delete this line and invert the conditional.